coderd/database/dbauthz/dbauthz.go

```diff
@@ -2209,7 +2209,28 @@ func (q *querier) InsertWorkspaceBuild(ctx context.Context, arg database.InsertW
 	}
 
 	if err = q.authorizeContext(ctx, action, w.WorkspaceBuildRBAC(arg.Transition)); err != nil {
-		return err
+		return xerrors.Errorf("authorize context: %w", err)
+	}
+
+	// If we're starting a workspace we need to check the template.
+	if arg.Transition == database.WorkspaceTransitionStart {
+		// Fetch the template. We have to ensure that the request
+		// abides by the update policy of the templates.
+		t, err := q.db.GetTemplateByID(ctx, w.TemplateID)
+		if err != nil {
+			return xerrors.Errorf("get template by id: %w", err)
+		}
+
+		// If the template requires the active version we need to check if
+		// the user is a template admin. If they aren't and are attempting
+		// to use a non-active version then we must fail the request.
+		if t.RequireActiveVersion {
+			if arg.TemplateVersionID != t.ActiveVersionID {
+				if err = q.authorizeContext(ctx, rbac.ActionUpdate, t); err != nil {
+					return xerrors.Errorf("cannot use non-active version: %w", err)
```

Comment:
This is also a little weird in that we're enforcing `RequireActiveVersion` regardless of whether there is a license of advanced template scheduling.

And, if the turn it on, then stop paying us and remove their license, we won't let them turn it off on any templates where they left it on.

Reply:
That's true, do we want to check entitlements here though?

Reply:
I'm not sure how best to do this tbh. The cleanest way I can think of is to move `schedule.TemplateScheduleStore` out of `schedule` and into `database` and then pass the `atomic.Pointer` that we use in `coderd` down into `dbauthz` so that it gets the updates from the `updateEntitlements` loop. 

That way we avoid introducing a dependency on `coderd/schedule` in `dbauthz` and we still get the benefits of license entitlements without introducing the complexity. Thoughts?